src/messageprocessor.ts

```diff
@@ -66,46 +78,51 @@ export class MessageProcessor {
+        let results = reg.exec(content);
         while (results !== null) {
-          const id = results[1];
-          const member = msg.guild.members.get(id);
-          const memberId = `@_discord_${id}:${this.opts.domain}`;
-          const memberStr = member ? member.user.username : memberId;
-          content = content.replace(results[0], memberStr);
-          results = USER_REGEX.exec(content);
+            const id = results[1];
+            const member = msg.guild.members.get(id);
+            const memberId = `@_discord_${id}:${this.opts.domain}`;
+            const memberStr = member ? member.user.username : memberId;
+            content = content.replace(results[0], memberStr);
+            results = reg.exec(content);
         }
         return content;
     }
 
-    public ReplaceChannels(content: string, msg: Discord.Message): string {
-        let results = CHANNEL_REGEX.exec(content);
+    public ReplaceChannels(content: string, msg: Discord.Message, postmark: boolean = false): string {
+        const reg = postmark ? CHANNEL_REGEX_POSTMARK : CHANNEL_REGEX;
+        let results = reg.exec(content);
         while (results !== null) {
-          const id = results[1];
-          const channel = msg.guild.channels.get(id);
-          const roomId = `#_discord_${msg.guild.id}_${id}:${this.opts.domain}`;
-          const channelStr = channel ? "#" + channel.name : "#" + id;
-          content = content.replace(results[0], `[${channelStr}](${MATRIX_TO_LINK}${roomId})`);
-          results = CHANNEL_REGEX.exec(content);
+            const id = results[1];
+            const channel = msg.guild.channels.get(id);
+            const roomId = `#_discord_${msg.guild.id}_${id}:${this.opts.domain}`;
+            const channelStr = channel ? "#" + channel.name : "#" + id;
+            content = content.replace(results[0], `[${channelStr}](${MATRIX_TO_LINK}${roomId})`);
+            results = reg.exec(content);
         }
         return content;
     }
 
-    public async ReplaceEmoji(content: string, msg: Discord.Message): Promise<string> {
-        let results = EMOJI_REGEX.exec(content);
+    public async ReplaceEmoji(content: string, msg: Discord.Message, postmark: boolean = false): Promise<string> {
+        const reg = postmark ? EMOJI_REGEX_POSTMARK : EMOJI_REGEX;
+        let results = reg.exec(content);
         while (results !== null) {
-          const id = results[1];
-          try {
-              const mxcUrl = await this.bot.GetGuildEmoji(msg.guild, id);
-              content = content.replace(results[0],
-                  `<img alt="${id}" src="${mxcUrl}" style="height: ${EMOJI_SIZE};"/>`);
-          } catch (ex) {
-              log.warn("MessageProcessor",
-              `Could not insert emoji ${id} for msg ${msg.id} in guild ${msg.guild.id}: ${ex}`,
-            );
-          }
-          results = EMOJI_REGEX.exec(content);
+            const animated = results[1] === "a";
+            const name = results[2];
```

Comment:
It gives linting errors here and the following line for "no magic numbers" but i have absolutely no idea how to fix that, as, well, that is how you fetch regex match groups